lib/tasks/registration_version.rake

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+namespace :registration_version do
+  desc "Migrates a Competition from V1 to V3"
+  task :migrate_v1_v3, [:competition_id] => [:environment] do |_, args|
+    competition_id = args[:competition_id]
+
+    if competition_id.blank?
+      abort "Competition id is required"
+    end
+
+    competition = Competition.find(competition_id)
+
+    if competition.nil?
+      abort "Competition #{competition_id} not found"
+    end
+
+    unless competition.registration_version_v1?
+      abort "Competition #{competition_id} is not on version 1"
+    end
+
+    LogTask.log_task("Migrating Registrations for Competition #{competition_id}") do
+      ActiveRecord::Base.transaction do
+        competition.registrations.each do |registration|
+          registration.update_column :competing_status, registration.compute_competing_status
+          registration.add_history_entry(registration.attributes.to_h, "rake task", "WST", "V2 Migration")
```

Comment:
I am 97% sure that `atrributes` is already a hash, so you shouldn't :tm: need `to_h` here.

Reply:
Also, what additional benefit does this history entry have?

Reply:
just giving a starting point for the registration history. I could also just use the accepted_at/accepted_by/deleted_at/deleted_by/created_at to create it?

Reply:
Yeah, since we have that data why not use it?